File: app/src/main/java/io/github/superbderrick/kotlinvideoplayer/VideoPlayer/VideoPlayerActivity.java
package io.github.superbderrick.kotlinvideoplayer.VideoPlayer;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.SurfaceHolder;
import android.view.SurfaceView;
import android.view.View;

import io.github.superbderrick.kotlinvideoplayer.R;

public class VideoPlayerActivity extends AppCompatActivity implements View.OnClickListener, SurfaceHolder.Callback{
    private SurfaceView mSurfaceView;
    private SurfaceHolder mSurfaceHolder;
    private VideoController mVideoController;
    private String mVideoPath = "https://bitdash-a.akamaihd.net/content/MI201109210084_1/m3u8s/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.m3u8";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_video_playback);

        mSurfaceView = (SurfaceView)findViewById(R.id.surfaceview);
        mSurfaceView.setOnClickListener(this);

        mSurfaceHolder= mSurfaceView.getHolder();
        mSurfaceHolder.addCallback(this);
    }

    @Override
    public void surfaceCreated(SurfaceHolder surfaceHolder) {
        mVideoController = new VideoController();

        mVideoController.openVideo(mVideoPath, mSurfaceHolder);
    }

    @Override
    public void surfaceChanged(SurfaceHolder surfaceHolder, int i, int i1, int i2) {

    }

    @Override
    public void surfaceDestroyed(SurfaceHolder surfaceHolder) {

    }

    @Override
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.surfaceview:
                mVideoController.handleVideo();
                break;
            default:

                break;
        }
    }
}

File: app/src/main/java/io/github/superbderrick/kotlinvideoplayer/VideoPlayer/VideoPlayerActivity.kt
package io.github.superbderrick.kotlinvideoplayer.VideoPlayer

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.SurfaceHolder
import android.view.SurfaceView
import android.view.View
import android.widget.ImageButton
import android.widget.LinearLayout
import android.widget.SeekBar
import android.widget.Toast
import io.github.superbderrick.kotlinvideoplayer.R

class VideoPlayerActivity : AppCompatActivity(), View.OnClickListener, SurfaceHolder.Callback, SeekBar.OnSeekBarChangeListener{

    private val LOG_TAG = "VideoPlayerActivity"

    private lateinit var mSurfaceView: SurfaceView
    private lateinit var mSurfaceHolder: SurfaceHolder
    private lateinit var mVideoController: VideoController
    private lateinit var mSeekbarVideo: SeekBar
    private var mUserSelectedPosition: Int = 0
    private var mUserIsSeeking: Boolean = false

    private lateinit var mVideoControllerView: LinearLayout
    private lateinit var mBackwardBtn: ImageButton
    private lateinit var mPlayBtn: ImageButton
    private lateinit var mForwardBtn: ImageButton

    private val mVideoPath: String = "http://cdn-fms.rbs.com.br/vod/hls_sample1_manifest.m3u8"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_video_playback)

        setUIComponents();

        mSurfaceView.setOnClickListener(this)
        mSeekbarVideo.setOnSeekBarChangeListener(this)

        mSurfaceHolder = mSurfaceView.holder
        mSurfaceHolder.addCallback(this)
    }

    private fun setUIComponents(){
        mSurfaceView = findViewById(R.id.surfaceview)
        mSeekbarVideo = findViewById(R.id.seekbar_video)

        mVideoControllerView = findViewById(R.id.videoController_view)

        mBackwardBtn = findViewById(R.id.backward_btn)
        mPlayBtn = findViewById(R.id.play_btn)
        mForwardBtn = findViewById(R.id.forward_btn)

        mBackwardBtn.setOnClickListener(this)
        mPlayBtn.setOnClickListener(this)
        mForwardBtn.setOnClickListener(this)
    }

    override fun surfaceCreated(surfaceHolder: SurfaceHolder?) {
        mVideoController = VideoController()
        mVideoController.openVideo(mVideoPath, surfaceHolder, mSeekbarVideo)
    }

    override fun surfaceChanged(p0: SurfaceHolder?, p1: Int, p2: Int, p3: Int) {

    }

    override fun surfaceDestroyed(p0: SurfaceHolder?) {

    }

    /*
    Seekbar callback
     */
    override fun onStartTrackingTouch(p0: SeekBar?) {
        mUserIsSeeking = true
    }

    override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
        if(fromUser){
            mUserSelectedPosition = progress;
        }
    }

    override fun onStopTrackingTouch(p0: SeekBar?) {
        mUserIsSeeking = false
        mVideoController.seekVideo(mUserSelectedPosition)
    }
    //Seekbar callback

    override fun onClick(view: View?) {
        if(view != null){
            when(view.id){
                R.id.surfaceview -> {
                    mVideoController.handleVideo()
                }

                R.id.backward_btn -> {

                }

                R.id.play_btn -> {

                }

                R.id.forward_btn -> {

                }
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        mVideoController.releaseVideo()
    }

    override fun onStop() {
        super.onStop()
        mVideoController.pauseVideo()
    }
}